sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryBlobRestClient.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Containers.ContainerRegistry
+{
+    internal partial class ContainerRegistryBlobRestClient
+    {
+        /// <summary>
+        /// Replace the default implementation to call RedirectPolicy.SetAllowAutoRedirect(message, true).
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="digest"></param>
+        /// <returns></returns>
+        internal HttpMessage CreateGetBlobRequest(string name, string digest)
+        {
+            var message = _pipeline.CreateMessage();
+            var request = message.Request;
```

Comment:
Was `message.BufferResponse = false;` removed intentionally?

Reply:
😮 no it wasn't. I'm not sure how that even happened, considering it was copy/pasted! Good catch.

Reply:
Cool, ok, I wasn't sure if something about the new behavior made that necessary.  But this kind of mistake that's super easy to make unintentionally is essentially why I'm concerned about manually overriding these methods.  We've done a lot of configuring of the generated layer via generator knobs, and moving to manual has the effect of taking away that as the approach for implementing the methods, now, and in the future.  It'd be easy for a new dev to see those customizations and think that's the right place to make changes, then head scratch for a long time regarding why those changes aren't getting picked up.  Or any updates to the generator to then not get picked up in a new build, etc.

Reply:
That's fair, we may want to wait and see if other libraries need this functionality before adding it to the generator.

I'd still like to see if we can find a solution other than overriding the generated methods for this library.

Reply:
The other option is to do this globally for the whole client at pipeline creation time. as discussed [here](https://github.com/Azure/azure-sdk-for-net/pull/34645#discussion_r1214804433).

Reply:
I agree it makes sense to keep the complexity out of the redirect policy until more than one library needs the functionality.  I would still like to have a way to handle this without overwriting the generated methods for ACR, though, even if that means adding complexity to its pipeline.  Would the recommendation be for us to add a policy to the pipeline to throw on a 3xx response from an unexpected operation?

Reply:
In my mind, the additional policy is equivalent to overriding generated code since this behavior could change if the swagger does.

I believe the easiest alternative is to enable redirects for the whole pipeline. Another alternative could be to construct 2 pipelines, one with redirects enabled and one without, and use them as appropriate for each operation.

Reply:
We'd construct another pipeline and rest client with options set to enable redirects, like we do [here](https://github.com/Azure/azure-sdk-for-net/blob/6ad5dbbb1b5240e1aa7d9bb5dfd912371d69ddac/sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryClient.cs#L81-L91C13).

then use the 2nd rest client (`_restClientWithRedirects`) in the client operations that require redirect support.

Reply:
Oh yeah, sold!  Where do I sign?

Reply:
done!